dhall-docs/src/Dhall/Docs/Store.hs

```diff
@@ -0,0 +1,44 @@
+{-| Utilities to interact with the dhall-docs home directory
+-}
+module Dhall.Docs.Store (getDocsHomeDirectory, makeHashForDirectory) where
+
+import Dhall.Crypto (SHA256Digest (..), sha256Hash)
+import Path         (Abs, Dir, Path)
+import Path.IO      (XdgDirectory (..))
+
+import qualified Codec.Archive.Tar       as Tar
+import qualified Codec.Archive.Tar.Entry as Tar.Entry
+import qualified Data.ByteString.Lazy
+import qualified Data.List
+import qualified Path
+import qualified Path.IO
+
+{-| Fetches the dhall-docs home directory. If @XDG_DATA_HOME@ env var is
+    defined, then @${XDG_DATA_HOME}/dhall-docs@ will be returned. Otherwise,
+    "${HOME}/.local/share/dhall-docs"
+-}
+getDocsHomeDirectory :: IO (Path Abs Dir)
+getDocsHomeDirectory = do
+    dir <- Path.IO.getXdgDir Path.IO.XdgData $ Path.parseRelDir "dhall-docs"
```

Comment:
I was confused by the use of `parseRelDir` on a literal here. I now understand hat the result is a `Maybe`:

http://hackage.haskell.org/package/path-io-1.6.0/docs/Path-IO.html#v:getXdgDir

I guess changing this to a quasiquote or TH might again give us trouble with `-Wunused-imports`?

Reply:
Probably it will give use those errors. But since that module will be really small, I think the change is worth to do.